source/common/upstream/logical_dns_cluster.cc

```diff
@@ -51,18 +59,26 @@ void LogicalDnsCluster::startResolve() {
   info_->stats().update_attempt_.inc();
 
   active_dns_query_ = dns_resolver_.resolve(
-      dns_address, [this, dns_address](
-                       std::list<Network::Address::InstanceConstSharedPtr>&& address_list) -> void {
+      dns_address, dns_lookup_family_,
+      [this, dns_address](
+          std::list<Network::Address::InstanceConstSharedPtr>&& address_list) -> void {
         active_dns_query_ = nullptr;
         log_debug("async DNS resolution complete for {}", dns_address);
         info_->stats().update_success_.inc();
 
         if (!address_list.empty()) {
-          // TODO(mattklein123): IPv6 support as well as moving port handling into the DNS
-          // interface.
-          Network::Address::InstanceConstSharedPtr new_address(
-              new Network::Address::Ipv4Instance(address_list.front()->ip()->addressAsString(),
-                                                 Network::Utility::portFromTcpUrl(dns_url_)));
+          // TODO(mattklein123): Move port handling into the DNS interface.
+          Network::Address::IpVersion version = address_list.front()->ip()->version();
+          Network::Address::InstanceConstSharedPtr new_address;
+          if (version == Network::Address::IpVersion::v4) {
```

Comment:
can you do switch here, so we make sure that compiler will complain in case enum values changes (with low probability though :))